### PR TITLE
Add Generators page to documentation

### LINKED ISSUE #897: Add Generators page to documentation
As an improvement to the documentation site, let's add a page that describes all of the generator configuration in one place.

**TODO**

- [x] Include item on how to generate a sidecar folder

Might you be up for writing this @Spone?

### PR DESCRIPTION
**Pull request by Spone**
Closes #897

### Summary

This PR adds a Generators page to the docs, and moves some content from the "Building ViewComponents" page to this new page.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## main
 
+* Add [Generators](https://viewcomponent.org/guide/generators.html) page to documentation
```
Comment (joelhawksley) [inline CHANGELOG.md:5]:
```suggestion
* Add [Generators](https://viewcomponent.org/guide/generators.html) page to documentation.
```

File: docs/guide/generators.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+layout: default
+title: Generators
+parent: Building ViewComponents
+---
+
+# Generators
+
+The generator accepts a component name and a list of arguments.
+
+This creates an `ExampleComponent` with `title` and `content` attributes:
```
Comment (joelhawksley) [inline docs/guide/generators.md:11]:
```suggestion
To create an `ExampleComponent` with `title` and `content` attributes:
```

File: docs/guide/generators.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      create  app/components/example_component.rb
+      invoke  test_unit
+      create    test/components/example_component_test.rb
+      invoke  erb
+      create    app/components/example_component.html.erb
+```
+
+## Options
+
+The component generator accepts several options to customize its behavior.
```
Comment (joelhawksley) [inline docs/guide/generators.md:25]:
```suggestion
```

File: docs/guide/generators.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+### Override the default template engine
```
Comment (joelhawksley) [inline docs/guide/generators.md:27]:
```suggestion
### Override template engine
```

File: docs/guide/generators.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+ViewComponent includes template generators for the `erb`, `haml`, and `slim` template engines and will default to the template engine specified in `config.generators.template_engine`.
```
Comment (joelhawksley) [inline docs/guide/generators.md:29]: 🤔 should we add a similar guide for all `config` options?

File: docs/guide/generators.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+bin/rails generate component Example title --template-engine slim
+
+      create  app/components/example_component.rb
+      invoke  test_unit
+      create    test/components/example_component_test.rb
+      invoke  slim
+      create    app/components/example_component.html.slim
+```
+
+### Override the default test framework
```
Comment (joelhawksley) [inline docs/guide/generators.md:41]:
```suggestion
### Override test framework
```

File: docs/guide/generators.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+By default, the `config.generators.test_framework` is used.
```
Comment (joelhawksley) [inline docs/guide/generators.md:43]:
```suggestion
By default, `config.generators.test_framework` is used.
```

File: docs/guide/generators.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+      create  app/components/example_component.rb
+      invoke  test_unit
+      create    test/components/example_component_test.rb
+      invoke  preview
+      create    test/components/previews/example_component_preview.rb
+      invoke  erb
+      create    app/components/example_component.html.erb
+```
+
+### Place the view and assets in a [sidecar directory](/guide/sidecar_assets.html)
```
Comment (joelhawksley) [inline docs/guide/generators.md:69]:
```suggestion
### Place the view in a [sidecar directory](/guide/sidecar_assets.html)
```

File: docs/guide/generators.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+### Skip collision check
+
+The generator prevents naming collisions with existing components. To skip this check and force the generator to run, use the `--skip-collision-check` or `--force` option.
+
+## Customizing the generator
+
+The best way to tailor the component generator to your project needs, is to create your own.
+
+See [view_component-contrib](https://github.com/palkan/view_component-contrib#installation-and-generating-generators) for a possible approach.
```
Comment (joelhawksley) [inline docs/guide/generators.md:100]: As we're going to upstream most of `view_component-contrib`, I'd rather we leave this out.

**Review by joelhawksley — APPROVED** [review]
🙇🏻 thanks, Hans! ❤️

File: docs/guide/generators.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
```
Comment (joelhawksley) [inline docs/guide/generators.md:98]:
```suggestion
```

File: docs/guide/generators.md (hunk 5 above)
Comment (Spone) [reply docs/guide/generators.md:27]: I was about to suggest that, yes!

Comment (joelhawksley) [reply docs/guide/generators.md:27]: Awesome! I think it might make sense as part of the API docs. I'll look into it ❤️

Comment (Spone) [reply docs/guide/generators.md:27]: Can you create an issue for this? I may be able to tackle it.

**Pull request merged**